students/Sahlberg/Lesson5/mailroom3/mailroom3.py

```diff
@@ -0,0 +1,112 @@
+"""Ian Sahlberg
+Assignment 5 mailroom3
+Python 210
+01/14/2019"""
+
+#for file dating
+import datetime as dt
+
+#Dictionary of donors name, dollar amount
+donor = {'bob johnson': [150.00],
+         'susan skoosan': [2000.00, 550.00],
+         'tim tam':[10.50],
+         'roxanne raffle':[13.00, 75.00, 123.00],
+         'jon jacob':[5000.00, 5.00]
+         }
+
+#Dictionary of user options
+user_select = {0:'1 - Send a Thank You', 1:'2 - Create a Report', 2:'4 - quit', 3:'3 - Send letters to everyone', 4:'Enter an option: List (to give a list of names), A donors name, \"quit\" , to quit.', 5:'How much did they donate?'}
+
+def donate(name, dictionary):
+    """Appends a donation amount to a given donor."""
+    amount = int(input(user_select.get(5)))
+    return amount#, dictionary[name].append(amount)
+
+def amount_append(name, dictionary, amount):
+    return dictionary[name].append(amount)
+
+def print_list(dictionary):
+    """Prints our the donor list"""
+    return print(list(dictionary.keys()))
+
+def user_options():
+    """Main user options menu choices."""
+    user_action = input('Please choose from the following options:  \n'+ user_select.get(0) + '\n'+ user_select.get(1) + '\n'+ user_select.get(3)+ '\n'+ user_select.get(2))
+    return user_action
+
+def email(name, amount):
+    """Create email for specific donor and amount donated."""
+    return 'Thank you {} for the generous donation of $ {:.2f}. We appreciate your generosity.\n\nSincerely, \n\nThe Helping R Us Team\n\n'.format(name.title(), amount)
+
+def quit():
+    """Quit the program"""
+    return print('Until next time!')
+
+def sort_and_stats_dict(dictionary):
+    """Create a stats dictionary from a donor dictionary and sort by total donated."""
+    pre_sort = {k: [sum(v), len(v), round((sum(v) / len(v)), 0)] for k, v in dictionary.items()}
+    print(pre_sort)
+    donor_sorted = sorted(list(pre_sort.items()), key=lambda x: x[1], reverse=True)
+    return donor_sorted
+
+def print_report(dictionary):
+    """Print out a report of donors and amounts donated, including stats (total donated, number of times donated, average donation)."""
+    print('Donor Name       |      Donor Total    |  Count of Donations    |  Average of Donations')
+    print('----------------------------------------------------------------------------------------')
+    count = 0
+    for donor in dictionary:
+        print(
+            f'{dictionary[count][0]:<20}$    {dictionary[count][1][0]:<23.2f}{dictionary[count][1][1]:<15}    ${dictionary[count][1][2]:>12.0f}')
+        count += 1
```

Comment:
a bit confused here - why is dictionary accessed by index? loop over dictionary `for key, value in a_dict` to get key and value right away.